lib/capistrano/tasks/deploy.rake

```diff
@@ -155,11 +155,20 @@ namespace :deploy do
 
       if valid.count >= fetch(:keep_releases)
         info t(:keeping_releases, host: host.to_s, keep_releases: fetch(:keep_releases), releases: valid.count)
-        directories = (valid - valid.last(fetch(:keep_releases)))
+        directories = (valid - valid.last(fetch(:keep_releases))).map do |release|
+          releases_path.join(release).to_s
+        end
+        begin
+          current_release = capture(:readlink, current_path).to_s
+          if directories.include?(current_release)
+            warn t(:wont_delete_current_release, host: host.to_s)
+            directories.delete(current_release)
+          end
+        rescue SSHKit::Command::Failed
```

Comment:
This worries me a bit. If the SSH execution fails for *any* reason, this `rescue` swallows it and prints a debug message. I assume this is being done to catch a failure for the `capture` on line 162 if it returns a non-zero exit status. Is there a more robust way to check the existence of the file without resorting to a `rescue`?

Reply:
That's correct. The idea that comes to mind is to do something like `capture("readlink #{path} || echo '' ")` (so it returns a empty string instead of the error), but that seems a bit hackish to me.

At the same time I'm not familiar enough with the project code to fully understand the risks of that rescue there. Is there another exception that would be safer to catch? Any other suggestions to improve this?

Reply:
The idiomatic way to do this in Capistrano is using the `test` command. So you could write:

```ruby
if test("[ -d #{current_path} ]")
  current_release = capture(:readlink, current_path).to_s
  # ...
else
  debug t(:no_current_release, host: host.to_s)
end
```

Reply:
@mattbrictson Awesome. Thanks very much for the suggestion!